### ISSUE TITLE
Potential dependency conflicts between oraclebmc and cryptography

### ISSUE THREAD

**Issue by NeolithEra** [oracle/oci-python-sdk#255]
Hi, as shown in the following full dependency graph of **_oci_**, **_oci_** requires **_cryptography ==2.8_**, **_oci_** requires **_pyopenssl >=17.5.0_** (**_pyopenssl 19.1.0_** will be installed, i.e., the newest version satisfying the version constraint), and directed dependency **_pyopenssl 19.1.0_** transitively introduces **_cryptography >=2.8_**.

Obviously, there are multiple version constraints set for **_cryptography_** in this project. However, according to pip's _“first found wins”_ installation strategy, **_cryptography 2.8_** (i.e., the newest version satisfying constraint **_==2.8_**) is the actually installed version.

Although the first found package version **_cryptography 2.8_** just satisfies the later dependency constraint （**_cryptography ==2.8_**), such installed version is very close to the upper bound of the version constraint of cryptography specified by **_pyopenssl 19.1.0_**. 

Once **_pyopenssl_** upgrades，its newest version will be installed, as **_oci_** does not specify the upper bound of version constraint for **_pyopenssl_**. Therefore, it will easily cause a dependency conflict (build failure), if the upgraded **_pyopenssl_** version introduces a higher version of **_cryptography_**, violating its another version constraint **_==2.8_**.

According to the release history of pyopenssl, it habitually upgrates Cryptography in its recent releases. For instance, pyopenssl 19.0.0 upgrated Cryptography’s constraint from >=2.2.1 to >=2.3,and pyopenssl 19.1.0 upgrated Cryptography’s constraint from >=2.3 to >=2.8. 

As such, it is a warm warning of a potential dependency conflict issue for oci.

### Dependency tree
```
oci - 2.14.2
| +- certifi(install version:2020.4.5.1 version range:*)
| +- configparser(install version:3.5.0 version range:==4.0.2)
| +- cryptography(install version:2.8 version range:==2.8)
| | +- asn1crypto(install version:1.3.0 version range:>=0.21.0)
| | +- idna(install version:2.9 version range:>=2.1)
| | +- six(install version:1.11.0 version range:>=1.4.1)
| +- pyopenssl(install version:19.1.0 version range:>=17.5.0)
| | +- cryptography(install version:2.8 version range:>=2.8)
| | | +- asn1crypto(install version:1.3.0 version range:>=0.21.0)
| | | +- idna(install version:2.9 version range:>=2.1)
| | | +- six(install version:1.11.0 version range:>=1.4.1)
| | +- six(install version:1.11.0 version range:>=1.5.2)
| +- python-dateutil(install version:2.5.3 version range:>=2.5.3,<3.0.0)
| | +- six(install version:1.11.0 version range:*)
| | +- six(install version:1.11.0 version range:>=1.5)
| +- pytz(install version:2016.10 version range:>=2016.10)
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Suggested Solution
1.	Loosen the version range of **_cryptography_** to be **_>=2.8_**.
2.	Remove your direct dependency **_cryptography_**, and use the **_cryptography_** transitively introduced by **_pyopenssl_**.
3.	Change your direct dependency **_pyopenssl_** to be **_>=17.5.0,<=19.1.0_**.
@dshelbyo  Which solution do you prefer, 1 ,2or 3?
Please let me know your choice. May I pull a request to solve this issue?

**Comment by bhagwatvyas** [comment]
Hi @NeolithEra, thanks for bringing this potential dependency issue to our attention. The reason that the cryptography dependency is set to v2.8 is we have tested with, and have compliance and security approval to use v2.8.

While we are reviewing our options regarding updating cryptography, I believe the safest way is to go with option 3 to solve the immediate problem. I would be happy to review your Pull Request.

**Comment by bhagwatvyas** [comment]
Hi @NeolithEra, I see that you made a commit regarding this to your fork. Pinging you to open a pull request as well :)

**Comment by bhagwatvyas** [comment]
Hi @NeolithEra, ping again :)

**Comment by zedabeta** [comment]
I don't want to poach, but would it be alright if I opened this pull request?

**Comment by bhagwatvyas** [comment]
Hi @zedabeta, we're shipping these changes with the next release tomorrow.

**Comment by zedabeta** [comment]
Awesome, thanks for letting me know

**Comment by bhagwatvyas** [comment]
Closing this issue as this is fixed in v2.20.0. Thanks @NeolithEra and @zedabeta for your help!
